src/jit/lsra.cpp

```diff
@@ -4070,6 +4075,15 @@ void LinearScan::buildRefPositionsForNode(GenTree*                  tree,
             useCandidates  = allRegs(registerType);
         }
 
+#ifdef _TARGET_ARM_
+        if (tree->OperIsPutArgStruct())
+        {
+            // get i-th candidate
+            currCandidates = genFindLowestReg(candidates);
```

Comment:
This relies on the fact that the arg regs are used in order. It would be good to add an assert that this is the same register as returned by `GetRegNumByIdx` (or `GetRegNumByIndex` if you change it).